### PR TITLE
Add support for MP23DB01HP

### PR DESCRIPTION
**Pull request by grobx**

### CODE REVIEW

File: src/pcm_lowlevel_u585ai.cpp [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+  RCC_ClkInitStruct.SYSCLKSource = RCC_SYSCLKSOURCE_PLLCLK;
+  RCC_ClkInitStruct.AHBCLKDivider = RCC_SYSCLK_DIV1;
+  RCC_ClkInitStruct.APB1CLKDivider = RCC_HCLK_DIV1;
+  RCC_ClkInitStruct.APB2CLKDivider = RCC_HCLK_DIV1;
+  RCC_ClkInitStruct.APB3CLKDivider = RCC_HCLK_DIV1;
+
+  if (HAL_RCC_ClockConfig(&RCC_ClkInitStruct, FLASH_LATENCY_4) != HAL_OK) {
+    Error_Handler();
+  }
+  __HAL_RCC_PWR_CLK_DISABLE();
```
Comment (fpistm) [inline src/pcm_lowlevel_u585ai.cpp]: I would not recommend to disable it as you don't know if someone else enable it before.

File: src/pcm_lowlevel_u585ai.cpp [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+  if (PCM.OnRecv != nullptr) {
+    PCM.OnRecv();
+  }
+}
+
+/**
+  * @brief System Clock Configuration
+  * @retval None
+  */
+void SystemClock_Config(void)
```
Comment (fpistm) [inline src/pcm_lowlevel_u585ai.cpp]: I guess this should be removed and update the default one in the variant.

File: src/pcm_lowlevel_u585ai.cpp [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+{
+  RCC_OscInitTypeDef RCC_OscInitStruct = {0};
+  RCC_ClkInitTypeDef RCC_ClkInitStruct = {0};
```
Comment (fpistm) [inline src/pcm_lowlevel_u585ai.cpp]: To remove warning.
```suggestion
  RCC_OscInitTypeDef RCC_OscInitStruct = {};
  RCC_ClkInitTypeDef RCC_ClkInitStruct = {};
```

**Review by fpistm — CHANGES_REQUESTED** [review]

File: src/pcm_lowlevel_u585ai.cpp (hunk 2 above)
Comment (grobx) [reply src/pcm_lowlevel_u585ai.cpp]: As far as I can tell you, this clock configuration is needed only when using this library.

File: src/pcm_lowlevel_u585ai.cpp (hunk 1 above)
Comment (grobx) [reply src/pcm_lowlevel_u585ai.cpp]: This code was not created by me, I have absolutely no idea of what it does.

Comment (cparata) [reply src/pcm_lowlevel_u585ai.cpp]: I think that we can remove line 277 without any issue

File: src/pcm_lowlevel_u585ai.cpp (hunk 2 above)
Comment (fpistm) [reply src/pcm_lowlevel_u585ai.cpp]: Hi @grobx 
main issue here is you redefine the `SystemClock_Config()` which prevents user to redefine it on its own.
So the weak  `SystemClock_Config()` available in the variant should be updated to add the required clock config you need.

File: src/pcm_lowlevel_u585ai.cpp (hunk 1 above)
Comment (fpistm) [reply src/pcm_lowlevel_u585ai.cpp]: As example, other libary enable the clk of the PWR domain and assume it is on while this library simply disable it.

File: src/pcm_lowlevel_u585ai.cpp (hunk 2 above)
Comment (fpistm) [reply src/pcm_lowlevel_u585ai.cpp]: Question: is the default  SystemClock_Config defined in the variant is correct and allow you to have this library working else we need to update it in the core?

Comment (grobx) [reply src/pcm_lowlevel_u585ai.cpp]: It works even without this system clock configuration. So I think we don't need to add to the variant.

Comment (fpistm) [reply src/pcm_lowlevel_u585ai.cpp]: Fine.

**Review by cparata — APPROVED** [review]

**Review by cparata — COMMENTED** [review]
I think we can proceed with the merge

File: src/pcm_lowlevel_u585ai.cpp [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+    PeriphClkInit.PLL3.PLL3M = 2;
+    PeriphClkInit.PLL3.PLL3N = 48;
+    PeriphClkInit.PLL3.PLL3P = 2;
+    PeriphClkInit.PLL3.PLL3Q = 25;
+    PeriphClkInit.PLL3.PLL3R = 2;
+    PeriphClkInit.PLL3.PLL3RGE = RCC_PLLVCIRANGE_0;
+    PeriphClkInit.PLL3.PLL3FRACN = 0;
+    PeriphClkInit.PLL3.PLL3ClockOut = RCC_PLL3_DIVQ;
+    if (HAL_RCCEx_PeriphCLKConfig(&PeriphClkInit) != HAL_OK) {
+      while (1);
```
Comment (fpistm) [inline src/pcm_lowlevel_u585ai.cpp]: Would recommend to use Error_Handler();
```suggestion
      Error_Handler();
```

**Review by fpistm — APPROVED** [review]
One minor remarks.

**Comment by grobx** [comment]
> I think we can proceed with the merge

Hi @cparata, I made the last changes as suggested by @fpistm. Let me know if there's something else to add.

**Comment by cparata** [comment]
> > I think we can proceed with the merge
> 
> Hi @cparata, I made the last changes as suggested by @fpistm. Let me know if there's something else to add.

It is fine for me. I proceed with the merge

**Pull request merged**